torch/csrc/autograd/saved_variable.h

```diff
@@ -102,7 +102,11 @@ class TORCH_API SavedVariable {
   // hooks are defined. They are set before pack_hook is called and used after
   // unpack_hook is called.
   std::shared_ptr<Node> grad_fn_;
-  std::weak_ptr<Node> grad_accumulator_;
+  // For the usual case where leaf tensors are the input, we expect its
```

Comment:
Is there any risk of memory leak here? Why was this a weak_ptr before?

Reply:
I don't think we'd be leaking memory, since the graph would've kept the grad_fn alive in the normal case anyway.

With that, there's no reason we couldn't have saved a strong ptr, though saving a weak_ptr is a good sanity check.

Reply:
Ok